core/pom.xml

```diff
@@ -32,14 +106,19 @@
     <!-- scope "test" -->
     <dependency>
       <groupId>org.junit.jupiter</groupId>
-      <artifactId>junit-jupiter</artifactId>
+      <artifactId>junit-jupiter-api</artifactId>
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>org.wiremock</groupId>
       <artifactId>wiremock</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpcore</artifactId>
+      <scope>test</scope>
```

Comment:
I would expect `compile` or `runtime` here.

Reply:
HttpClient5 is compile, HttpClient4 is used tests like for `org.apache.http.HttpStatus.SC_BAD_REQUEST`

Reply:
IMO we should avoid mixing Http client versions in tests, if possible. E.g. use `import org.apache.hc.core5.http.HttpStatus;` instead. Maybe give it a try?

Reply:
<details><summary>With the change above we would break the "connectivity-oauth" dependency</summary>

![image](https://github.com/user-attachments/assets/5aa47de4-b2ef-41d6-aa37-c0ddce903d0c)


</details>

Reply:
Nevermind

<details><summary>"Effective POM" results in correct scope</summary>

![image](https://github.com/user-attachments/assets/33f0f2e9-c5ce-45b3-bcf4-0e786eaffbd3)

</details>